be/src/storage/lake/vacuum.cpp

```diff
@@ -347,12 +347,13 @@ static Status collect_files_to_vacuum(TabletManager* tablet_mgr, std::string_vie
     }
     auto t1 = butil::gettimeofday_ms();
     g_metadata_travel_latency << (t1 - t0);
-
-    *vacuumed_version = final_retain_version;
     if (!skip_check_grace_timestamp) {
         // All tablet metadata files encountered were created after the grace timestamp, there were no files to delete
+        // The vacuumed_version will be set to min garbage version
```

Comment:
the deleter is asynchronized and in batches, it just means the deletion is submit but may not success at all. However if reported the min vacuum version to the "supposed success version", so it may mislead FE not sending any vacuum requests further to clean again?

So the vacuumed_version should be set to the version that are firmly deleted in the system by this vacuum check, but not necessary includes the versions that are to be deleted?

Reply:
you are right.